enterprise/server/scheduling/task_router/task_router.go

```diff
@@ -284,6 +284,29 @@ func (tr *taskRouter) MarkComplete(ctx context.Context, action *repb.Action, cmd
 	log.Debugf("Preferred executor host ID %q added to %q", executorHostID, routingKey)
 }
 
+// MarkFailed clears the routing table after a task fails. This makes it so
+// that subsequent executions will run on random execution nodes.
+func (tr *taskRouter) MarkFailed(ctx context.Context, action *repb.Action, cmd *repb.Command, remoteInstanceName, executorHostID string) {
```

Comment:
nit: `use log.Ctx` funcs to make debugging easier

Reply:
👍 done